spec-main/api-ipc-renderer-spec.ts

```diff
@@ -0,0 +1,198 @@
+import { expect } from 'chai'
+//import * as http from 'http'
+import * as path from 'path'
+import { ipcMain, /*webContents,*/ BrowserWindow, WebContents, WebPreferences, webContents } from 'electron'
+import { emittedOnce } from './events-helpers'
+import { closeWindow } from './window-helpers';
+
+describe('ipcRenderer module', () => {
+  const fixtures = path.join(__dirname, '..', 'spec', 'fixtures')
+
+  let w: BrowserWindow
+  before(async () => {
+    w = new BrowserWindow({ show: false, webPreferences: { nodeIntegration: true } })
+    await w.loadURL('about:blank')
+  })
+  after(async () => {
+    await closeWindow(w)
+    w = null as unknown as BrowserWindow
+  })
+
+  describe('send()', () => {
+    it('should work when sending an object containing id property', async () => {
+      const obj = {
+        id: 1,
+        name: 'ly'
+      }
+      w.webContents.executeJavaScript(`{
+        const { ipcRenderer } = require('electron')
+        ipcRenderer.send('message', ${JSON.stringify(obj)})
+      }`)
+      const [, received] = await emittedOnce(ipcMain, 'message')
+      expect(received).to.deep.equal(obj)
+    })
+
+    it('can send instances of Date as ISO strings', async () => {
```

Comment:
Isn't it sending a Date object not an ISO string?

Reply:
That's just how our IPC works 🤷‍♂ see https://github.com/electron/electron/pull/20214 for the PR that will make this send a real Date :)